betel/data_set_builder.py

```diff
@@ -0,0 +1,69 @@
+import pathlib
+import abc
+from typing import Dict
+import pandas as pd
+from betel import utils
+from betel import info_files_helpers
+
+
+class DataSetBuilder(metaclass=abc.ABCMeta):
+    """A class for splitting data into train-validation-test sets."""
+
+    _RANDOM_SEED = 2579  # seed for random splitting
+
+    def __init__(self, input_dir: pathlib.Path, storage_dir: pathlib.Path):
+        """"Constructor.
+
+        :param input_dir: data to be split (output of the scraper)
+        :param storage_dir: storage directory for split data sets
+        """
+        self._storage_dir = storage_dir
+        self._storage_dir.mkdir(exist_ok=True, parents=True)
+
+        self._input_dir = input_dir
+        self._input_file = input_dir / utils.SCRAPER_INFO_FILE_NAME
+
+    def split(self, data: pd.DataFrame) -> Dict[str, pd.DataFrame]:
+        """Randomly splits a list of data into 3
+        (train-validation-test).
+
+        :param data: the list to be split
+        :return: the train-validation-test split in Dict format
+        """
+        app_list = self._sort(data)
+        app_list = app_list.sample(frac=1, random_state=self._RANDOM_SEED)
+
+        index_1 = int(0.7 * len(app_list))
```

Comment:
One more suggestion, it might be useful to make the train-test-validation split proportions configurable.